ocaml/xapi/xapi_cluster_host.ml

```diff
@@ -230,6 +197,49 @@ let enable ~__context ~self =
         handle_error error
     )
 
+(* Enable cluster_host in client layer via clusterd *)
+let resync_localhost ~__context =
+  let host = Helpers.get_localhost ~__context in
+  match find_cluster_host ~__context ~host with
+  | None      -> () (* no clusters exist *)
+  | Some self ->    (* cluster_host and cluster exist *)
+    let body = Printf.sprintf "Unable to create cluster host on %s."
+        (Db.Host.get_name_label ~__context ~self:host) in
+    let obj_uuid = Db.Host.get_uuid ~__context ~self:host in
+
+    try
+      (* If we have just joined, enable will prevent concurrent clustering ops *)
+      if not (Db.Cluster_host.get_joined ~__context ~self)
+      then join_internal ~__context ~self
+      else
+        if Db.Cluster_host.get_enabled ~__context ~self then begin
+          (* [enable] unconditionally invokes low-level enable operations and is idempotent.
+             RPU reformats partition, losing service status, never re-enables clusterd *)
+          debug "Cluster_host %s is enabled, starting up xapi-clusterd" (Ref.string_of self);
+          Xapi_clustering.Daemon.enable ~__context;
+
+          (* Note that join_internal and enable both use the clustering lock *)
+          enable ~__context ~self
+          end
+    with err ->
+      Backtrace.is_important err;
+      let body = Printf.sprintf "Error: %s\nMessage: %s" ExnHelper.(string_of_exn err) body in
+      Xapi_alert.add ~msg:Api_messages.cluster_host_enable_failed ~cls:`Host ~obj_uuid ~body;
+      raise err
+
+let resync_host ~__context ~host =
+  let localhost = Helpers.get_localhost ~__context in
+  if not (host = localhost) then begin
```

Comment:
i don't really want to include a host parameter here, but if i understand correctly i have to because it needs to be included in order for somebody to resync another host

that is, the forwarding should have been done already in messaging_forwarding.ml

Reply:
It is better to double check that we made the changes correctly, perhaps it should be an `assert` instead of an `if` statement though.
You do need a host object to query the cluster host object on, but as above, that can always be just the local host. The functions in `xapi_host` take a `host` or `self` parameter too, so lets keep it.

Reply:
this should no longer be necessary, provided `pool_resync` is calling `resync_host` correctly!